### ISSUE TITLE
SUGGESTION: A WAY TO TRIGGER ACTUAL CAPS LOCK

### ISSUE THREAD

**Issue by mnsc** [ililim/dual-key-remap#30]
MAYBE SOMETHING LIKE HOLDING DOWN CAPSLOCK FOR ONE MINUTE! AND TO TURN IT OFF, HOLD IT ANOTHER MINUTE!

THANKS!

#CAPSLOCKDAY

**Comment by Pitasi** [comment]
Yeah it happened to me to press the caps lock key before dual-key-remap is started on logon and then I can't disable it as the caps lock key now behave as esc/ctrl. Right now I have to go through task manager, kill dual-key-remap, press caps lock, start dual-key-remap again. Super annoying.

It would also be great if there's a way for dual-key-remap to disable the caps lock before starting.

**Comment by mnsc** [comment]
Haha, I love it that my jokingly created issue for caps-lock-day provoked a legitimate feature request. But I would argue that the solution maybe should be to have a clear way of shutting down instead of task killing if something isn't working as expected? Rather than a feature to trigger caps lock.

**Comment by mnsc** [comment]
And today I was playing around with automated ui tests and somehow got caps lock triggered! =D I had to taskkill and restart like @Pitasi 🤷

**Comment by Pitasi** [comment]
You can easily get caps lock triggered by:
- launching an app as admin
- launching dual-key-remap NOT as admin
- pressing caps lock while the admin app is focused

There is a workaround for this explained in another issue, can't remember which one :)

**Comment by ililim** [comment]
Yeah this is kind annoying and something I'd like to fix one way or another.

For better or worse I tried to make this tool somewhat flexible, so it doesn't actually know that it's being used for capslock rebinding (it rebinds any key to a dual-functionality key). So can't hold it to disable capslock, as the app doesn't know it's capslock that you're rebinding!

Quick and easy way to turn off caps: Open task manager (Ctrl + Shift + Esc). Press the capslock key (dual-key-remap won't work in admin programs). Should have capslock toggled.

I agree, re: it is tedious to close the app. I would indeed like to add a taskbar icon – last time I checked it wasn't a super trivial addition, if anyone has any open source examples of how to implement this it could help kickstart this.

What you can also do is use Sharp Keys as described in the readme: https://github.com/ililim/dual-key-remap#rebind-keys-during-login. This rebinds caps to escape permanently. Then use this tool toggle the key between ctrl and escape for the dual key functionality. Fixes it not working in admin programs also (well, partially).

The longer term game for this app is to make it actually run under admin privileges. Maybe even support some of sharp key's functionality. The big blocker here is figuring out how to have an elevated app startup as soon as windows does – I was playing around with scheduled background tasks but that didn't work for me during testing.

**Comment by escape0707** [comment]
As I mainly use this app to remap <kbd>Caps Lock</kbd> to <kbd>Ctrl</kbd> and <kbd>Esc</kbd>, I don't have to press <kbd>Esc</kbd> anymore. So I remapped <kbd>Esc</kbd> to <kbd>Caps Lock</kbd>.

This config is:
```javascript
remap_key=CAPSLOCK
when_alone=ESCAPE
with_other=CTRL

remap_key=ESCAPE
when_alone=CAPSLOCK
with_other=ESCAPE
```

It works fine most of the time and for most applications like `explorer.exe` and `firefox` but I found there is some problem

**Comment by mnsc** [comment]
This joke is now played out for me at least. I appreciate the workarounds, especially the, in hindsight obvious, remap esc to capslock. But my muscle memory will reach for esc "as esc" way more often than this tool have left me with caps lock on. So I will close this issue now and leave it to others to create issues for taskbar icon/running elevated which I personally doesn't miss that much.
